crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -314,20 +514,27 @@ impl AuthorityStorePruner {
                 }
             });
         }
+
         tokio::task::spawn(async move {
             loop {
                 tokio::select! {
-                    _ = prune_interval.tick(), if config.num_epochs_to_retain != u64::MAX => {
+                    _ = objects_prune_interval.tick(), if config.num_epochs_to_retain != u64::MAX => {
                         if let Err(err) = Self::prune_objects_for_eligible_epochs(&perpetual_db, &checkpoint_store, &objects_lock_table, config, metrics.clone(), indirect_objects_threshold).await {
                             error!("Failed to prune objects: {:?}", err);
                         }
                     },
+                    _ = checkpoints_prune_interval.tick(), if config.num_epochs_to_retain_for_checkpoints.is_some() => {
+                        if let Err(err) = Self::prune_checkpoints_for_eligible_epochs(&perpetual_db, &checkpoint_store, &objects_lock_table, config, metrics.clone(), indirect_objects_threshold).await {
```

Comment:
it is a little bit weird we have to pass `indirect_objects_threshold` even though it's only used for object pruning